packages/manager/modules/nutanix/src/dashboard/general-info/redeploy/controller.js

```diff
@@ -0,0 +1,132 @@
+import {
+  REDEPLOY_INITIAL_CONFIG,
+  REDEPLOY_CUSTOM_CONFIG,
+  PRISM_CENTRAL_TYPES,
+  IPV4_REGEX,
+  IPV4_BLOCK_REGEX,
+  TRACKING_PREFIX,
+} from './constants';
+
+export default class NutanixGeneralInfoRedeployCtrl {
+  /* @ngInject */
+  constructor($translate, atInternet, Validator) {
+    this.$translate = $translate;
+    this.atInternet = atInternet;
+    this.Validator = Validator;
+    this.REDEPLOY_INITIAL_CONFIG = REDEPLOY_INITIAL_CONFIG;
+    this.REDEPLOY_CUSTOM_CONFIG = REDEPLOY_CUSTOM_CONFIG;
+    this.prismCentralTypes = PRISM_CENTRAL_TYPES;
+    this.IPV4_REGEX = IPV4_REGEX;
+    this.IPV4_BLOCK_REGEX = IPV4_BLOCK_REGEX;
+  }
+
+  $onInit() {
+    this.initalConfig = angular.copy(this.cluster.targetSpec);
```

Comment:
Nitpick: better to use lodash or plain javascript methods to clone the object. angular.copy might create a issue during upgrade.